packages/databricks-vscode/src/configuration/auth/orchestrate.ts

```diff
@@ -31,17 +29,43 @@ export async function orchestrate<S, KEYS extends string>(
     steps: Record<KEYS, Step<S, KEYS>>,
     start: KEYS,
     maxSteps = 20,
-    logger?: logging.NamedLogger
+    logger?: logging.NamedLogger,
+    cancellationToken?: CancellationToken
 ): Promise<S> {
     let counter = 0;
 
     let step: KEYS | undefined = start;
-    while (step && steps[step]) {
+    function isCancelled() {
+        return cancellationToken?.isCancellationRequested === true;
+    }
+    while (step && steps[step] && !isCancelled()) {
         counter += 1;
         if (counter > maxSteps) {
             throw new OrchestrationLoopError();
         }
-        const result: StepResult<S, KEYS> = await steps[step]();
+
+        let result: StepResult<S, KEYS> | undefined = undefined;
+        const task: Promise<StepResult<S, KEYS> | undefined> = (async () => {
+            return await steps[step!]();
+        })().catch((e) => {
+            if (!isCancelled()) {
+                throw e;
+            }
+            return undefined;
+        });
```

Comment:
A bit hard to read this part, might be simpler to just set the task here (`const task: Promise<StepResult<S, KEYS> | undefined> = steps[step]();`), and do the try/catch around `Promise.race`. What do you think?

Reply:
I want to have a listener on the task in case it it is preempted. Otherwise, unhandled promises cause errors to show up in the extension host logs.